util/embedders.py

```diff
@@ -1,36 +1,51 @@
+from typing import Optional
 from embedders.classification.count_based import (
     BagOfCharsSentenceEmbedder,
     BagOfWordsSentenceEmbedder,
     TfidfSentenceEmbedder,
 )
-from embedders.classification.contextual import TransformerSentenceEmbedder
+from embedders.classification.contextual import OpenAISentenceEmbedder, HuggingFaceSentenceEmbedder, CohereSentenceEmbedder
 from embedders.extraction.count_based import BagOfCharsTokenEmbedder
 from embedders.extraction.contextual import TransformerTokenEmbedder
 from embedders.classification.reduce import PCASentenceReducer
 from embedders.extraction.reduce import PCATokenReducer
 from embedders import Transformer
+from submodules.model import enums
 
 from submodules.model.business_objects import record
 
 
 def get_embedder(
     project_id: str,
     embedding_type: str,
-    config_string: str,
     language_code: str,
+    platform: str,
+    model: Optional[str] = None,
+    api_token: Optional[str] = None,
 ) -> Transformer:
-    if embedding_type == "classification":
+    if embedding_type == enums.EmbeddingType.ON_ATTRIBUTE.value:
         batch_size = 128
         n_components = 64
-        if config_string == "bag-of-characters":
-            return BagOfCharsSentenceEmbedder(batch_size=batch_size)
-        elif config_string == "bag-of-words":
-            embedder = BagOfWordsSentenceEmbedder(batch_size=batch_size)
-        elif config_string == "tf-idf":
-            embedder = TfidfSentenceEmbedder(batch_size=batch_size)
-        else:
-            embedder = TransformerSentenceEmbedder(
-                config_string=config_string, batch_size=batch_size
+        if platform == "python":
+            if model == "bag-of-characters":
+                return BagOfCharsSentenceEmbedder(batch_size=batch_size)
+            elif model == "bag-of-words":
+                embedder = BagOfWordsSentenceEmbedder(batch_size=batch_size)
+            elif model == "tf-idf":
+                embedder = TfidfSentenceEmbedder(batch_size=batch_size)
```

Comment:
since we only have these three options I think we should throw something on else

Reply:
solved